client/my-sites/checkout/checkout-thank-you/licensing-thank-you-auto-activation.tsx

```diff
@@ -0,0 +1,240 @@
+		hasProductInfo ? getProductName( state, productSlug ) : null
+	);
+
+	const isProductListFetching = useSelector( ( state ) => getIsProductListFetching( state ) );
+	const userName = useSelector( getCurrentUserName );
+	const jetpackSites = useSelector( getJetpackSites ) as JetpackSite[];
+
+	const supportTicketStatus = useSelector( ( state ) =>
+		getJetpackCheckoutSupportTicketStatus( state, receiptId )
+	);
+
+	const supportContactLink =
+		'https://jetpack.com/support/install-jetpack-and-connect-your-new-plan/';
+
+	const [ selectedSite, setSelectedSite ] = useState( '' );
+	const [ error, setError ] = useState< TranslateResult | false >( false );
+
+	const onContinue = useCallback(
+		( e ) => {
+			e.preventDefault();
+			if ( selectedSite === 'activate-license-manually' ) {
+				const manualActivationUrl = addQueryArgs(
+					{
+						receiptId,
+						source,
+						jetpackTemporarySiteId,
+					},
+					`/checkout/jetpack/thank-you/licensing-manual-activate/${ productSlug }`
+				);
+				return page( manualActivationUrl );
+			}
+			dispatch(
+				recordTracksEvent( 'calypso_siteless_checkout_submit_website_address', {
+					product_slug: productSlug,
+					site_url: selectedSite,
+					receipt_id: receiptId,
+				} )
+			);
+			dispatch(
+				requestUpdateJetpackCheckoutSupportTicket(
+					selectedSite,
+					receiptId,
+					source,
+					jetpackTemporarySiteId
+				)
+			);
+		},
+		[ selectedSite, dispatch, productSlug, receiptId, source, jetpackTemporarySiteId ]
+	);
+
+	useEffect( () => {
+		if ( supportTicketStatus === 'success' ) {
+			const thankYouCompletedUrl = addQueryArgs(
+				{
+					siteId: jetpackTemporarySiteId,
+					receiptId,
+				},
+				`/checkout/jetpack/thank-you-completed/no-site/${ productSlug }`
+			);
+			page( thankYouCompletedUrl );
+		} else if ( supportTicketStatus === 'failed' ) {
+			setError(
+				translate( 'There was a problem submitting your website address, please try again.' )
+			);
+		}
+	}, [ jetpackTemporarySiteId, receiptId, supportTicketStatus, productSlug, translate ] );
+
+	const siteSelectOptions =
+		jetpackSites &&
+		jetpackSites.map( ( site: JetpackSite ) => ( {
+			value: site?.URL,
+			label: site.URL,
+			props: {
+				key: site.ID,
+				selected: site.URL === selectedSite,
+				value: site.URL,
+				onClick: () => {
+					setSelectedSite( site.URL );
+				},
+			},
+		} ) );
+
+	const lastSelectOption = {
+		value: 'activate-license-manually',
+		label: translate( "I don't see my site. Let me configure it manually" ),
+		props: {
+			key: 'activate-license-manually',
+			selected: 'activate-license-manually' === selectedSite,
+			value: 'activate-license-manually',
+			onClick: () => {
+				setSelectedSite( 'activate-license-manually' );
+			},
+		},
+	};
+	const selectDropdownItems = siteSelectOptions && [
+		...siteSelectOptions,
+		...[ lastSelectOption ],
+	];
+	const selectedItem = find( selectDropdownItems, 'props.selected' );
```

Comment:
I don't really know if using `lodash` here will increase the size of the bundle, but to be on the safe side, I suggest replacing `lodash.find` with the JavaScript native version of `find`.

So,
```
const selectedItem = find( selectDropdownItems, 'props.selected' );
```

would become

```
const selectedItem = selectDropdownItems.find( ( item ) => item.props.selected );
```

Reply:
Yes right, good call! 🙂